### PR TITLE
Multiple small fixes

### PR DESCRIPTION
**Pull request by brandond**
#### Proposed Changes ####

Multiple small 1/2 line fixes, one per commit:
1. Disable containerd experimental snapshot labels
    Related to #2455 and containerd/containerd#4684
    These were not meant to be enabled by default, break images with many layers, and will be disabled by default on the next containerd release.
1. Fix containerd sock path warning
    Resolves warning 1 from #2471
1. Add Cluster ID support to k3s stub cloud controller
    Resolves warning 2 from #2471.
    As per https://github.com/kubernetes/cloud-provider/issues/12 the ClusterID requirement was never really followed through on, so the flag is probably going to be removed in the future.
1. Disable configure-cloud-routes and external service/route programming support when using k3s stub cloud controller
    Resolves warning 3 from #2471
1. Respect --data-dir path for crictl.yaml
    Related to rancher/rke2#474
    Note that anyone who customizes the data-dir path will have to set CRI_CONFIG_FILE to the correct path when using the wrapped binaries (crictl, etc). This is better than dropping files in the incorrect location.
1. Fix size check and make script shell consistent
    Related to #2205

#### Types of Changes ####

* Bugfix

#### Verification ####

1. Pull image listed in https://github.com/rancher/k3s/issues/2455#issuecomment-720635256, confirm no error message.
1. Look for log message listed in #2471 part 1
1. Look for log message listed in #2471 part 2
1. Look for log message listed in #2471 part 3
1. Start k3s with --data-dir, confirm that no files are created in /var/lib/rancher/k3s
1. Build k3s or check CI logs, verify no errors while checking binary size.

#### Linked Issues ####

See issues linked above

#### Further Comments ####

One commit per issue, hopefully that's easier to review. I am not planning on squashing this before merging since the commits should all stand on their own.

### CODE REVIEW

File: scripts/manifest [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
```
Comment (briandowns) [inline scripts/manifest:1]: What are we gaining with this change?

Comment (brandond) [reply scripts/manifest:1]: Consistency across all shell scripts in the scripts/ directory:
```
brandond@dev01:/var/lib/repos/go/src/github.com/rancher/k3s$ head -n 1 scripts/* | grep -F '#!' | sort | uniq -c
     29 #!/bin/bash
      3 #!/bin/sh
      1 #!/usr/bin/env bash
```

File: pkg/agent/config/config.go [hunk 2]
```diff
@@ -461,7 +461,7 @@ func get(envInfo *cmds.Agent, proxy proxy.Proxy) (*config.Node, error) {
 	}
 
 	if !nodeConfig.Docker && nodeConfig.ContainerRuntimeEndpoint == "" {
-		nodeConfig.AgentConfig.RuntimeSocket = nodeConfig.Containerd.Address
+		nodeConfig.AgentConfig.RuntimeSocket = "unix://" + nodeConfig.Containerd.Address
```
Comment (dweomer) [inline pkg/agent/config/config.go:464]: if the user specifies a url scheme on the cli this is going to cause problems, no?

File: pkg/agent/run.go [hunk 3]
```diff
@@ -58,9 +57,9 @@ func setupCriCtlConfig(cfg cmds.Agent, nodeConfig *daemonconfig.Node) error {
 		}
 	}
 
-	agentConfDir := datadir.DefaultDataDir + "/agent/etc"
+	agentConfDir := filepath.Join(cfg.DataDir, "agent", "etc")
 	if _, err := os.Stat(agentConfDir); os.IsNotExist(err) {
-		if err := os.MkdirAll(agentConfDir, 0755); err != nil {
+		if err := os.MkdirAll(agentConfDir, 0700); err != nil {
```
Comment (dweomer) [inline pkg/agent/run.go:62]: /etc/ is typically 0755 which this emulates. granted that the parent is 0700 this change doesn't amount to much

File: pkg/agent/run.go [hunk 4]
```diff
@@ -141,12 +140,12 @@ func Run(ctx context.Context, cfg cmds.Agent) error {
 		}
 	}
 
-	cfg.DataDir = filepath.Join(cfg.DataDir, "agent")
-	if err := os.MkdirAll(cfg.DataDir, 0700); err != nil {
+	agentDir := filepath.Join(cfg.DataDir, "agent")
+	if err := os.MkdirAll(agentDir, 0700); err != nil {
```
Comment (dweomer) [inline pkg/agent/run.go:144]: we've replaced an in-place mutation with a local alias. the intentional side effect will no longer take place. this assumes that the side effect was intentional

File: pkg/agent/config/config.go (hunk 2 above)
Comment (brandond) [reply pkg/agent/config/config.go:464]: No, this is only used if the ContainerRuntimeEndpoint is an empty string ie not specified by the user.

File: pkg/agent/templates/templates.go [hunk 5]
```diff
@@ -34,6 +34,7 @@ const ContainerdConfigTemplate = `
 
 {{- if .NodeConfig.AgentConfig.Snapshotter }}
 [plugins.cri.containerd]
+  disable_snapshot_annotations = true
```
Comment (dweomer) [inline pkg/agent/templates/templates.go:37]: where is this coming from?

File: pkg/agent/run.go (hunk 3 above)
Comment (brandond) [reply pkg/agent/run.go:62]: I've tested this and it doesn't seem to be a problem, and is now consistent with the other agent dir that we create below.

File: manifests/ccm.yaml [hunk 6]
```diff
@@ -78,3 +78,6 @@ subjects:
 - kind: User
   name: cloud-controller-manager
   namespace: kube-system
+- kind: ServiceAccount
```
Comment (galal-hussein) [inline manifests/ccm.yaml:81]: Why do we add service account cloud-controller-manager to the rbac?

File: pkg/cloudprovider/cloudprovider.go [hunk 7]
```diff
@@ -58,5 +58,5 @@ func (k *k3s) ProviderName() string {
 }
 
 func (k *k3s) HasClusterID() bool {
-	return false
+	return true
```
Comment (dweomer) [inline pkg/cloudprovider/cloudprovider.go:61]: @galal-hussein is this reversing an intentional decision?

File: pkg/agent/run.go (hunk 4 above)
Comment (brandond) [reply pkg/agent/run.go:144]: Correct. In my reading of the code the mutation to DataDir was incorrect; the only places that needed the modified paths have been updated to use the local var instead.

File: pkg/agent/templates/templates.go (hunk 5 above)
Comment (brandond) [reply pkg/agent/templates/templates.go:37]: It's covered in the PR description and commit message:
> Disable containerd experimental snapshot labels
> Related to #2455 and containerd/containerd#4684
> These were not meant to be enabled by default, break images with many layers, and will be disabled by default on the next containerd release.

File: pkg/cloudprovider/cloudprovider.go (hunk 7 above)
Comment (brandond) [reply pkg/cloudprovider/cloudprovider.go:61]: As per kubernetes/cloud-provider#12 the ClusterID requirement was never really followed through on, so the flag is probably going to be removed in the future. Setting the flag and returning false here currently does nothing except cause a warning to be printed (I checked).

File: manifests/ccm.yaml (hunk 6 above)
Comment (brandond) [reply manifests/ccm.yaml:81]: Changing the cloud-provider args made the built-in cloud-controller-manager want to watch Nodes, which previously lacked permission to.

File: pkg/cloudprovider/cloudprovider.go (hunk 7 above)
Comment (galal-hussein) [reply pkg/cloudprovider/cloudprovider.go:61]: yeah clusterID i believe  was only utilized by aws provider thats why we returned false, I am good with the change if it doesnt generate unwanted warnings/errors when starting the ccm

**Review by dweomer — APPROVED** [review]
LGTM :+1:  (apologies for not reading the original pr comment)

**Review by briandowns — APPROVED** [review]

**Pull request merged**